src/imagery/i.eodag/i.eodag.py

```diff
@@ -788,13 +910,17 @@ def print_eodag_queryables(**kwargs):
         "Annotated",
     ]  # For testing and catching edge cases
 
+    # Possible types to be passed through the query option
+    # TODO: We can possibly extend the supported types
+    SUPPORTED_TYPES = ["str", "int", "float", "Literal"]
+
     def get_type(info):
         potential_type = info.__args__[0]
         if potential_type.__name__ != "Optional":
-            assert potential_type.__name__ in types_options
+            assert potential_type.__name__ in METADATA_TYPES
```

Comment:
assert in Python code like this doesn't do what you think it does. It is in fact useless, at it is not guaranteed to be executed. It can be disabled, or not present in some installations. If the condition checked is required to be checked for correct execution of the logic, it should be written as such, like using an if.
Many Python linters will try to warn about this.


You can see here 
https://docs.python.org/3.12/reference/simple_stmts.html#the-assert-statement that this syntax
`assert_stmt ::=  "assert" expression ["," expression]`
, ie `assert 1==2, "that expression should be false"`

Is equivalent to
```python
if __debug__:
    if not expression1: raise AssertionError(expression2)
```

ie: 

```python
if __debug__:
    if not 1==2:
        raise AssertionError("that expression should be false")
```


You can also read

https://dbader.org/blog/catching-bogus-python-asserts


Or the ruff rule S101 https://docs.astral.sh/ruff/rules/assert/

Reply:
I didn't know that about the assert statements. Thanks for the references, @echoix!
I will fix this problem.